src/serialization/compact/DefaultCompactWriter.ts

```diff
@@ -425,3 +429,50 @@ export class DefaultCompactWriter implements CompactWriter {
         }
     }
 }
+
+/**
+ * Checks that the Compact serializable array items that are written are of
+ * a single type.
+ */
+export class SingleTypeCompactArrayItemChecker<T> {
+    
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    private clazz: Function;
+    
+    public check(value: T): void {
+        const clazzType = value.constructor;
```

Comment:
I think we should check for the undefined and null elements. They don't have a constructor field. They will throw exception here. Writing null in an array of compact field is completely legal, though it should not be undefined. Using undefined will throw an exception but I think we should handle that as well. We can throw an exception here.

Also note that, the object created with "Object.create(null)" (the null object) has its constructor undefined. For the null object, the serializer won't be able to be found, and it will fail "No serializer is found" error in CompactStreamSerializer

We should write tests of these.

Reply:
We talked about this and we agree on adding new controls and throws. And also I wrote new tests related.